source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,177 @@
+* Enable [SSH](https://www.synology.com/en-global/knowledgebase/DSM/help/DSM/AdminCenter/system_terminal) service
+* Optionally start Home Assistant on bootup of your Synology NAS
 
-SSH onto your synology & login as admin or root
+## {% linkable_title Making new Python 3 package %}
 
- - Log in with your own administrator account
- - Switch to root using:
+#### {% linkable_title Installing Docker on [Mint, Ubuntu or derivatives](https://docs.docker.com/install/linux/docker-ce/ubuntu/) %}
 
+Install Docker:
 ```bash
-$ sudo -i
+# sudo apt-get update
+# sudo apt-get install docker.io
+```
+Start Docker and enable start on system bootup:
+```bash
+# sudo systemctl start docker
+# sudo systemctl enable docker
 ```
 
+#### {% linkable_title Preparing compiling environment %}
 
-Check the path to python3 (assumed to be /volume1/@appstore/py3k/usr/local/bin)
-
+Install Git:
+```bash
+# sudo apt-get install git
+```
+Fork and clone [spksrc](https://github.com/SynoCommunity/spksrc), this will make a directory "~/spksrc":
 ```bash
-# cd /volume1/@appstore/py3k/usr/local/bin
+git clone https://github.com/SynoCommunity/spksrc.git ~/spksrc
 ```
 
-Install PIP (Python's package management system)
+Edit the following 2 files.
 
-```bash
-# ./python3 -m ensurepip
+Add these edits to make cross compiled Python module packages which Home Assistant needs to install and enable usage of the "[Cloud](/components/cloud/)" and [Homekit](/components/homekit/)" components. Most components will setup just fine, but certain components need a python package that compiles on installation, which will fail as Synology did not provide a compiler, causing said need for cross compilation.<br/>
+Edit "*~/spksrc/spk/python3/src/requirements.txt*" and add the following text to the end of the file:
 ```
+## It may happen you want to add a component to Home Assistant, but it fails to install a Python package.
+## In the logs you see a error "Unable to install package *Module_Name*", if you install that
+# package with pip, some of its required packages fail with error "Failed building wheel for *Module_Name*"
+## Add these Python modules/packages that fail with that "building wheel" error in to the list below.
 
-Use PIP to install Homeassistant package 0.64.3
+# Example format of module:
+#pythonmodule==version
 
-```bash
-# ./python3 -m pip install homeassistant==0.64.3
+## Cross compilation requirements for installing Home Assistant (Tested to work on 84.3).
+cffi==1.11.5
+bcrypt==3.1.4
+cryptography==2.3.1
+
+## Cross compilation requirement for installing "Warrant" module (Needed by "Cloud" Component)
+pycryptodome==3.7.2
+
+## Cross compilation requirement for installing "HAP_python" module (Needed by "Homekit" Component)
+curve25519-donna==1.3
+ed25519==1.4
+```
+
+Add these edits to fix OpenSSL errors when using the "[Xiaomi_Aqara](/components/Xiaomi_Aqara/)" component.<br/>
```

Comment:
Add these edits to fix OpenSSL errors when using the [Xiaomi_Aqara](/components/Xiaomi_Aqara/) component.<br/>